source/on.js

```diff
@@ -0,0 +1,31 @@
+import curryN from './internal/_curryN';
+
+
+/**
+ * Takes a binary function `f`, a unary function `g`, and two values.
+ * Applies `g` to each value, then applies the result of each to `f`.
+ *
+ * Also known as the P combinator.
+ *
+ * @func
+ * @memberOf R
+ * @category Function
+ * @sig ((a, a) -> b) -> (c -> a) -> c -> c -> b
```

Comment:
I'd prefer `((b, b) -> c) -> (a -> b) -> a -> a -> c`. I like the types to flow from `a` to `b` to `c`.